tornado/test/httpserver_test.py

```diff
@@ -834,37 +834,53 @@ def setUp(self):
         super().setUp()
         self.tmpdir = tempfile.mkdtemp()
         self.sockfile = os.path.join(self.tmpdir, "test.sock")
-        sock = netutil.bind_unix_socket(self.sockfile)
         app = Application([("/hello", HelloWorldRequestHandler)])
         self.server = HTTPServer(app)
-        self.server.add_socket(sock)
-        self.stream = IOStream(socket.socket(socket.AF_UNIX))
-        self.io_loop.run_sync(lambda: self.stream.connect(self.sockfile))
+        if sys.platform.startswith("linux"):
+            self.sockabstract = "\0" + os.path.basename(self.tmpdir)
```

Comment:
This is a clever way to get a unique name for the abstract socket in most cases, although it does have some weaknesses since the abstract socket namespace may be shared more widely than the filesystem (e.g. containers with host networking, or just processes with different values of TMPDIR. So I'd rather use a uuid4 here than tie the abstract socket name to the temp directory.

Reply:
Switched to generating a unique abstract namespace identifier using [uuid4()](https://docs.python.org/3/library/uuid.html#uuid.uuid4) in https://github.com/tornadoweb/tornado/pull/3405/commits/be9e153b36fe3161394ebdf58c6d4aa426836821.